File: src/main/java/org/apache/prepbuddy/datacleansers/ReplacementFunction.java
package org.apache.prepbuddy.datacleansers;

import org.apache.prepbuddy.utils.Replacement;

import java.io.Serializable;

public class ReplacementFunction implements Serializable {
    private Replacement[] replacements;

    public ReplacementFunction(Replacement... replacements) {
        this.replacements = replacements;
    }

    public String replace(String current) {
        for (Replacement replacement : replacements) {
            if (replacement.matches(current)) return replacement.replacementValue();
        }
        return current;
    }
}

File: python/py_prep_buddy/transformers/replacementfunction.py
class ReplacementFunction(object):
    def __init__(self, gateway, func):
        self.__gateway = gateway
        self.__func = func

    def replace(self, row_record):
        return self.__func(row_record)

    class Java:
        implements = ["spark_context.org.apache.prepbuddy.transformers.ReplacementFunction"]
